src/IB/IBFEInstrumentPanel.cpp

```diff
@@ -0,0 +1,1067 @@
+                {
+                    U[k] += v(i_c, k) * wgt;
+                }
+            }
+        }
+#if (NDIM == 3)
+    }
+#endif
+    return U;
+} // linear_interp
+
+Vector
+linear_interp(const Vector& X,
+              const Index<NDIM>& i_cell,
+              const Vector& X_cell,
+              const SideData<NDIM, double>& v,
+              const Index<NDIM>& /*patch_lower*/,
+              const Index<NDIM>& /*patch_upper*/,
+              const double* const /*x_lower*/,
+              const double* const /*x_upper*/,
+              const double* const dx)
+{
+#if !defined(NDEBUG)
+    TBOX_ASSERT(v.getDepth() == 1);
+#endif
+    Vector U(Vector::Zero());
+    for (unsigned int axis = 0; axis < NDIM; ++axis)
+    {
+        boost::array<bool, NDIM> is_lower;
+        for (unsigned int d = 0; d < NDIM; ++d)
+        {
+            if (d == axis)
+            {
+                is_lower[d] = false;
+            }
+            else
+            {
+                is_lower[d] = X[d] < X_cell[d];
+            }
+        }
+#if (NDIM == 3)
+        for (int i_shift2 = (is_lower[2] ? -1 : 0); i_shift2 <= (is_lower[2] ? 0 : 1); ++i_shift2)
+        {
+#endif
+            for (int i_shift1 = (is_lower[1] ? -1 : 0); i_shift1 <= (is_lower[1] ? 0 : 1); ++i_shift1)
+            {
+                for (int i_shift0 = (is_lower[0] ? -1 : 0); i_shift0 <= (is_lower[0] ? 0 : 1); ++i_shift0)
+                {
+                    const Vector X_side(X_cell[0] + (static_cast<double>(i_shift0) + (axis == 0 ? -0.5 : 0.0)) * dx[0],
+                                        X_cell[1] + (static_cast<double>(i_shift1) + (axis == 1 ? -0.5 : 0.0)) * dx[1]
+#if (NDIM == 3)
+                                        ,
+                                        X_cell[2] + (static_cast<double>(i_shift2) + (axis == 2 ? -0.5 : 0.0)) * dx[2]
+#endif
+                                        );
+                    const double wgt =
+                        (((X[0] < X_side[0] ? X[0] - (X_side[0] - dx[0]) : (X_side[0] + dx[0]) - X[0]) / dx[0]) *
+                         ((X[1] < X_side[1] ? X[1] - (X_side[1] - dx[1]) : (X_side[1] + dx[1]) - X[1]) / dx[1])
+#if (NDIM == 3)
+                         *
+                         ((X[2] < X_side[2] ? X[2] - (X_side[2] - dx[2]) : (X_side[2] + dx[2]) - X[2]) / dx[2])
+#endif
+                             );
+                    const Index<NDIM> i(i_shift0 + i_cell(0),
+                                        i_shift1 + i_cell(1)
+#if (NDIM == 3)
+                                            ,
+                                        i_shift2 + i_cell(2)
+#endif
+                                            );
+                    const SideIndex<NDIM> i_s(i, axis, SideIndex<NDIM>::Lower);
+                    U[axis] += v(i_s) * wgt;
+                }
+            }
+#if (NDIM == 3)
+        }
+#endif
+    }
+    return U;
+} // linear_interp
+}
+
+//**********************************************
```

Comment:
Please drop comments proceeding each of the function implementations. It is enough to say what the functions do in the header file.